tests/migration.go

```diff
@@ -80,34 +81,52 @@ func ConfirmVMIPostMigration(virtClient kubecli.KubevirtClient, vmi *v1.VirtualM
 	return vmi
 }
 
-func SetDedicatedMigrationNetwork(nad string) *v1.KubeVirt {
+func setOrClearDedicatedMigrationNetwork(nad string, set bool) *v1.KubeVirt {
 	virtClient, err := kubecli.GetKubevirtClient()
 	Expect(err).ToNot(HaveOccurred())
 
 	kv := util.GetCurrentKv(virtClient)
 
-	if kv.Spec.Configuration.MigrationConfiguration == nil {
-		kv.Spec.Configuration.MigrationConfiguration = &v1.MigrationConfiguration{
-			Network: &nad,
+	// Saving the list of virt-handler pods prior to changing migration settings, see comment below.
+	listOptions := metav1.ListOptions{LabelSelector: v1.AppLabel + "=virt-handler"}
+	virtHandlerPods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(), listOptions)
+	Expect(err).ToNot(HaveOccurred(), "Failed to list the virt-handler pods")
+
+	if set {
+		if kv.Spec.Configuration.MigrationConfiguration == nil {
+			kv.Spec.Configuration.MigrationConfiguration = &v1.MigrationConfiguration{}
+		}
+		kv.Spec.Configuration.MigrationConfiguration.Network = &nad
+	} else {
+		if kv.Spec.Configuration.MigrationConfiguration != nil {
+			kv.Spec.Configuration.MigrationConfiguration.Network = nil
 		}
 	}
 
-	kv.Spec.Configuration.MigrationConfiguration.Network = &nad
-
-	return UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
-}
+	res := UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
 
-func ClearDedicatedMigrationNetwork() *v1.KubeVirt {
-	virtClient, err := kubecli.GetKubevirtClient()
-	Expect(err).ToNot(HaveOccurred())
+	// By design, changing migration settings trigger a re-creation of the virt-handler pods, amongst other things.
+	//   However, even if SetDedicatedMigrationNetwork() calls UpdateKubeVirtConfigValueAndWait(), VMIs still seem to get scheduled on outdated virt-handler pods.
+	//   Not sure if it's a bug or a feature though. Waiting for all "old" virt-handlers to disappear ensures test VMIs will be created on updated virt-handler pods.
```

Comment:
This is not a bug. `UpdateKubeVirtConfigValueAndWait` is only waiting for the new config to be propagated but that doesn't guarantee that operator had time to do the job.